File: 2346-largest-3-same-digit-number-in-string/largest-3-same-digit-number-in-string.cpp
class Solution {
public:
    string largestGoodInteger(string num) {
        vector<string> vals = {"999", "888", "777", "666", "555", "444", "333", "222", "111", "000"};
        for (string s: vals) if (num.find(s) != string::npos) return s;
        return "";
    }
};

File: 2346-largest-3-same-digit-number-in-string/largest-3-same-digit-number-in-string.java
class Solution {
    public String largestGoodInteger(String num) {
        String res = "", cur = "000";
        for (int i=0; i<num.length()-2; i++) {
            if (num.charAt(i) == num.charAt(i+1) && num.charAt(i+1) == num.charAt(i+2) && Integer.parseInt(num.substring(i, i+3)) >= Integer.parseInt(cur)) {
                res = num.substring(i, i+3);
                cur = num.substring(i, i+3);
            }
        }
        return res;
    }
}